src/drivers/intel/cavs/ipc.c

```diff
@@ -93,6 +109,8 @@ static void ipc_irq_handler(void *arg)
 		/* mask Busy interrupt */
 		ipc_write(IPC_DIPCCTL, dipcctl & ~IPC_DIPCCTL_IPCTBIE);
 
+		increment_ipc_received_counter();
+
```

Comment:
How about those "dropping msg" happened in line 117? At the case handled_counter will be never equal to recceived_counter?

Reply:
@keyonjie 
As I understand correctly in our current IPC flow, we do not have any IPC's queuing. Driver can send new IPC only when firmware accomplishes previous ipc task and triggers interrupt to host. 
It happens in 177 line:
```
ipc->host_pending = 0;
/* write 1 to clear busy, and trigger interrupt to host*/
#if CAVS_VERSION == CAVS_VERSION_1_5
	ipc_write(IPC_DIPCT, ipc_read(IPC_DIPCT) | IPC_DIPCT_BUSY);
```
So in current flow we will never have situation that the variable ```_ipc->host_pending``` will be set to true in line 117.